test/WebJobs.Script.Tests/ExtensionBundleManagerTests.cs

```diff
@@ -0,0 +1,301 @@
+            return new ExtensionBundleManager(bundleOptions, environment, MockNullLoggerFactory.CreateLoggerFactory());
+        }
+
+        private TestEnvironment GetAppServiceEnvironment()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            string downloadPath = string.Empty;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                environment.SetEnvironmentVariable(AzureWebsiteHomePath, "D:\\home");
+            }
+            else
+            {
+                environment.SetEnvironmentVariable(AzureWebsiteHomePath, "//home");
+            }
+            return environment;
+        }
+
+        private ExtensionBundleOptions GetTestExtensionBundleOptions(string id, string version)
+        {
+            var options = new ExtensionBundleOptions
+            {
+                Id = id,
+                Version = VersionRange.Parse(version, true),
+                DownloadPath = _downloadPath
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                options.ProbingPaths.Add(@"C:\Program Files (x86)\FuncExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle");
+            }
+            else
+            {
+                options.ProbingPaths.Add("/FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle");
+                options.ProbingPaths.Add("/home/site/wwwroot/.azureFunctions/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle");
+            }
+
+            return options;
+        }
+
+        private Tuple<Mock<IFileSystem>, Mock<DirectoryBase>, Mock<FileBase>> CreateFileSystem()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            var fileBase = new Mock<FileBase>();
+            var dirBase = new Mock<DirectoryBase>();
+            fileSystem.SetupGet(f => f.Directory).Returns(dirBase.Object);
+            fileSystem.SetupGet(f => f.File).Returns(fileBase.Object);
+            return new Tuple<Mock<IFileSystem>, Mock<DirectoryBase>, Mock<FileBase>>(fileSystem, dirBase, fileBase);
+        }
+
+        public void Dispose()
+        {
+            FileUtility.Instance = null;
+        }
+
+        private class MockHttpHandler : HttpClientHandler
+        {
+            private HttpStatusCode _statusCodeForIndexJson;
+            private HttpStatusCode _statusCodeForZipFile;
+
+            public MockHttpHandler(HttpStatusCode statusCodeForIndexJson, HttpStatusCode statusCodeForZipFile)
+            {
+                _statusCodeForIndexJson = statusCodeForIndexJson;
+                _statusCodeForZipFile = statusCodeForZipFile;
+            }
+
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                await Task.Delay(1);
```

Comment:
Can you clarify? May want to just use `Task.Yield` instead

Reply:
Just used it to make Task.Dely. Task.Yield works as well.